pyutils/pymcutil.py

```diff
@@ -1,98 +1,288 @@
+        pdg = (data['trkmcsim', 'pdg'])
+        mask = (pdg == 11) & (rank == 0)
+        self.logger.log(f"Returning mask for is_electron", "success")
+        return mask
+    except Exception as e:
+        self.logger.log(f"Exception in is_electron: {e}", "error")
+        return None
+
+  def is_positron(self, data):
+    """ return mask for all events with true e+n """
+    try:
+        rank = data['trkmcsim', 'rank']
+        pdg = (data['trkmcsim', 'pdg'])
+        mask = (pdg == -11) & (rank == 0)
+        self.logger.log(f"Returning mask for is_positron", "success")
+        return mask
+    except Exception as e:
+        self.logger.log(f"Exception in is_positron: {e}", "error")
+        return None
         
-        Args:
-            data : array in format produced from processing
-        """
-        self.particle_count_return = None
-        self.print_prefix = "[pymcutil] "
+  def is_particle(self, data, code):
+    """ returns true if the trkmcsim has pdg code for chosen particle """
+    try:
+        mask = (data['trkmcsim']['pdg'] == int(code))
+        self.logger.log(f"Returning mask for is_particle", "success")
+        return mask
+    except Exception as e:
+        self.logger.log(f"Exception in is_particle: {e}", "error")
+        return None
+
+
+  def start_process(self, data, process):
+    """ returns true if the trkmcsim has process code for chosen start code """
+    try:
+        mask = (data['trkmcsim', 'startCode'] == int(process))
+        self.logger.log(f"Returning mask for start_process", "success")
+        return mask
+    except Exception as e:
+        self.logger.log(f"Exception in start_process: {e}", "error")
+        return None
+
+  def stop_process(self, data, process):
+    """ returns true if the trkmcsim has process code for chosen stop code"""
+    try:
+        mask = (data['trkmcsim', 'stopCode'] == int(process))
+        self.logger.log(f"Returning mask for stop_process", "success")
+        return mask
+    except Exception as e:
+        self.logger.log(f"Exception in stop_process: {e}", "error")
+        return None
         
-        print(f"{self.print_prefix}Initialised")
-      
-    def count_particle_types(self, data):
-      """
-      Counts the occurrences of different particle types based on
-      simulation data, leveraging the properties of Awkward Arrays.
+  def is_CeMinusEndpoint(self, data):
+    """ returns true if the trkmcsim has process code for ce- endpoint generator """
+    try:
+        mask = (data['trkmcsim', 'startCode'] == 167)
+        self.logger.log(f"Returning mask for is_CeMinusEndpoint", "success")
+        return mask
+    except Exception as e:
+        self.logger.log(f"Exception in is_CeMinusEndpoint: {e}", "error")
+        return None
 
-      Args:
-          data (ak.Array): An Awkward Array containing simulation data,
-                           including ''trkmcsim' nested field.
+  def is_CeMinusLeadingLog(self, data):
```

Comment:
Is there a reason that you don't use the ```start_process``` function here to avoid the repeated try-catch block?